buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -183,34 +208,69 @@ synchronized void start() {
             spec.from(distroArtifact.resolve("config").toFile());
             spec.into(configFile.getParent());
         });
-        configure();
-        startElasticsearchProcess(distroArtifact);
+
+        try {
+            createWorkingDir(distroArtifact);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        createConfiguration();
+
+        plugins.forEach(plugin -> runElaticsearchBinScript(
+            "elasticsearch-plugin",
+            "install", "--batch", plugin.toString())
+        );
+
+        startElasticsearchProcess();
+    }
+
+    private void runElaticsearchBinScript(String tool, String... args) {
+        services.loggedExec(spec -> {
+            spec.setEnvironment(getESEnvironment());
+            spec.workingDir(workingDir);
+            if (OperatingSystem.current().isWindows()) {
```

Comment:
As mentioned above, here's an example of relying on Gradle's implementation of OS detection. We should probably come to an agreement on how to standardize this.

Reply:
Good catch, i  missed this one. I'll move the  OS detection out to it's own class and replace it here